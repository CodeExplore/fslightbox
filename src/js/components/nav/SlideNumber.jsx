import React, { useState } from 'react';

const SlideNumber = (
    {
        fsLightbox: {
            componentsStates: {
                slideNumberUpdater: slideNumberUpdaterState
            },
            data: { sourcesCount },
            stageIndexes
        }
    }
) => {
    const [currentSlideNumberUpdaterValue, setSlideNumberUpdaterValue] = useState(false);
    slideNumberUpdaterState.get = () => currentSlideNumberUpdaterValue;
    slideNumberUpdaterState.set = setSlideNumberUpdaterValue;

    return (
        <div className="fslightbox-slide-number-container fslightbox-flex-centered">
            <div>{ stageIndexes.current + 1 }</div>
            <div className="fslightbox-slash">/</div>
            <div>{ sourcesCount }</div>
        </div>
    );
};
export default SlideNumber;
