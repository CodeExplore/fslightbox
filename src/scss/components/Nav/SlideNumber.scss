.fslightbox-slide-number-container {
  height: 100%;
  font-size: 14px;
  color: #d7d7d7;
  z-index: 0;
  max-width: 46px;

  div {
    padding: 0 2px;
  }
}

.fslightbox-slash {
  padding-top: .15rem;
}