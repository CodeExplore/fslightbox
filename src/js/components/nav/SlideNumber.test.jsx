import React from 'react';
import { shallow } from "enzyme";
import SlideNumber from "./SlideNumber";
import { testComponentStateForStateChainAndFsLightbox } from "../../../tests/__tests-helpers__/testComponentStateForStateChainAndFsLightbox";

const fsLightbox = {
    componentsStates: {
        slideNumberUpdater: {}
    },
    data: {
        sourcesCount: 4
    },
    stageIndexes: {
        current: 0
    }

};
let slideNumber = shallow(<SlideNumber fsLightbox={ fsLightbox }/>);

testComponentStateForStateChainAndFsLightbox('slideNumberUpdater', fsLightbox);

test('SlideNumber dom', () => {
    expect(slideNumber).toMatchSnapshot();
});
