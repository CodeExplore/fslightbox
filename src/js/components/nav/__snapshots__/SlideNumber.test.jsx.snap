// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SlideNumber DOM sourcesCount > 1 static DOM should match snapshot 1`] = `
<div
  className="fslightbox-slide-number-lightboxContainer fslightbox-flex-centered"
>
  <div>
    7
  </div>
  <div
    className="fslightbox-slash"
  >
    /
  </div>
  <div>
    5
  </div>
</div>
`;

exports[`SlideNumber dom 1`] = `
<div
  className="fslightbox-slide-number-container fslightbox-flex-centered"
>
  <div>
    1
  </div>
  <div
    className="fslightbox-slash"
  >
    /
  </div>
  <div>
    4
  </div>
</div>
`;

exports[`SlideNumber dom should match snapshot 1`] = `
<div
  className="fslightbox-slide-number-container fslightbox-flex-centered"
>
  <div>
    1
  </div>
  <div
    className="fslightbox-slash"
  >
    /
  </div>
  <div>
    4
  </div>
</div>
`;
